pkg/jetstream/message_handler.go

```diff
@@ -0,0 +1,115 @@
+package jetstream
+
+import (
+	"context"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+type handleFunc = func(ctx context.Context, msg jetstream.Msg, output chan *message.Message)
+
+// handleMsg provides the core processing logic for a single JetStream message
+func (s *Subscriber) handleMsg(ctx context.Context, msg jetstream.Msg, output chan *message.Message) {
+	m, err := unmarshal(msg)
+	if err != nil {
+		s.logger.Error("cannot unmarshal message", err, nil)
+	}
+
+	messageLogFields := watermill.LogFields{
+		"ID":              m.UUID,
+		"ReceivedSubject": msg.Subject(),
+		"Len":             len(msg.Data()),
+	}
+
+	s.logger.Info("got msg", messageLogFields)
+
+	ctx, cancelCtx := context.WithCancel(ctx)
+	m.SetContext(ctx)
+	defer cancelCtx()
+
+	timeout := time.NewTimer(s.ackWait)
```

Comment:
hello, I think this is not entirely correct because the ackWait time does not start when the message handler receives the message, but rather when the nats library gets this message from the server (which might be a different time due to pre-buffering). For a more detailed explanation of why I think this can be a problem check this issue: https://github.com/nats-io/nats.go/issues/1929